pageserver/src/tenant/storage_layer/split_writer.rs

```diff
@@ -481,7 +481,7 @@ mod tests {
 
     #[tokio::test]
     async fn write_split_discard() {
-        write_split_helper("split_writer_write_split_discard", false).await;
+        write_split_helper("split_writer_write_split_discard", true).await;
```

Comment:
So we set `discard=true` here, but are we asserting anything about the resulting behavior?  Is this test meant to be checking a particular behavior?

Reply:
Yes. If we discard all, the writer returns the persistent key that gets discarded. We check if the key ranges and lsn ranges are correct.